libafl_bolts/src/ownedref.rs

```diff
@@ -147,6 +218,7 @@ impl<'a, T: Sized> AsMut<T> for OwnedRefMut<'a, T> {
     #[must_use]
     fn as_mut(&mut self) -> &mut T {
         match self {
+            OwnedRefMut::RefRaw(r, _) => unsafe { r.as_mut().unwrap() },
```

Comment:
This PR is definitely of an improvement, in that we're now storing a raw pointer rather than a reference - this is important because holding a reference while the memory is mutated is UB.

However, it looks to me like it's still pretty easy to end up with UB by calling this method on the return value of `ShMem::as_owned_ref_mut_of`. `as_{ref,mut}` should probably be marked `unsafe` and have a comment stating that the memory can't be mutated (or even read, in the case of `as_mut`) while the returned reference is live, and point out that such mutation can happen via `ShMem`.

Reply:
My idea was that every possible constructor of `OwnedRef::RefRaw` is marked `unsafe` so that we don't need to mark all as unsafe. We cannot mark every _use_ of the OwnedRef unsafe, since we want to implement the `Serialize` trait - and marking all of the serialization functions unsafe is out of our control.

Reply:
This is the reason I added `UnsafeMarker` - there is no legal way to create a `RefRaw` without calling an `unsafe` constructor first. That's the best we can do I fear.

Reply:
> My idea was that every possible constructor of OwnedRef::RefRaw is marked unsafe so that we don't need to mark all as unsafe.

I see what you're saying. It's perhaps worth noting that the standard library takes the opposite approach - it's always safe to create raw pointers, always `unsafe` to dereference them.

Reply:
If we wouldn't need to serialize/deserialize it would be the way to go, but it's not an option here, sadly

Reply:
Do you mean we should add a comment to the Safety section here or something?